### ISSUE TITLE
Right aligning text

### ISSUE THREAD

**Issue by exuvo** [Noxagonal/Vulkan2DRenderer#142]
I got some text showing using vk2dInstance->GetResourceManager()->LoadFontResource and GenerateTextMesh but it would be really nice to either have a draw right aligned flag to GenerateTextMesh or a cheap way to calculate text width before hand to manually position the GenerateTextMesh call correctly.

I'm pretty sure just summing up the horizantal advance and kerning in a loop is enough but maybe you know of a better way?
```
for( auto c : text ) {
	auto gi = fi->GetGlyphInfo( font_face, c );
	location.x	+= gi->horisontal_advance + kerning;
}
```

-edit-
It would also be nice if the mesh returned by GenerateTextMesh had a field for width used so you can easily keep writing horizontally on the same line.

**Comment by Noxagonal** [comment]
That is more or less how you get the width, just without kerning for the last text character. However I would like to return a Rect2f for the total size instead of the width. Also needs a check for vertical text.

I'll add aabb for Mesh today that should work until I implement the cheaper calculation.

**Comment by Noxagonal** [comment]
Added AABB to vk2d::Mesh, this will work for the time being.

I'll keep this issue open until I implement a function to get the font "would be size".

**Comment by Noxagonal** [comment]
Added vk2d::FontResource::CalculateRenderedSize().
This will get the exact AABB area as if it was rendered.

**Comment by exuvo** [comment]
This does not seem to return correct values for me.

Writing the string "2100-001 00:00:00" using LiberationSans-Regular.ttf at 18 glyph_texel_size no scaling, visually occupies ca 150x20 pixels on my screen.

But the aabb on the text mesh gives me [[-507, -12], [10, 374]], GetAreaSize=[517, 386]. I'm writing a single line at the bottom left corner of the window so i don't get why both width and height are so large. I'm using center of screen is 0,0.

After some debugging it seems to be because you preload ret.aabb with a value that does not include the render location. So aabb starts at ex top_left = {x = 0, y = -12}, bottom_right = {x = 10, y = 0}. But then when writing first character `tcoords			= ( coords + location )` and the min,max:es keep that initial bottom_right.x = 10.

I'll send a patch with a fix.

**Comment by Noxagonal** [comment]
Seems like location also got scaled if text had scaling.
Fixed in ed576b3
Should work from now on but double check text locations.
